src/_pytest/pathlib.py

```diff
@@ -955,21 +955,25 @@ def scandir(
 
     The returned entries are sorted according to the given key.
     The default is to sort by name.
+    If the directory does not exist, return an empty list.
     """
-    entries = []
-    with os.scandir(path) as s:
-        # Skip entries with symlink loops and other brokenness, so the caller
-        # doesn't have to deal with it.
-        for entry in s:
-            try:
-                entry.is_file()
-            except OSError as err:
-                if _ignore_error(err):
-                    continue
-                raise
-            entries.append(entry)
-    entries.sort(key=sort_key)  # type: ignore[arg-type]
-    return entries
+    try:
```

Comment:
I would prefer a narrow try/except as done in https://github.com/pytest-dev/pytest/pull/13084, can you please update to use a similar solution?

Reply:
> I would prefer a narrow try/except as done in https://github.com/pytest-dev/pytest/pull/13084, can you please update to use a similar solution?

Yes, of course